src/test/test_depends_on.py

```diff
@@ -0,0 +1,105 @@
+from alembic_utils.pg_materialized_view import PGMaterializedView
+from alembic_utils.pg_view import PGView
+from alembic_utils.replaceable_entity import register_entities
+from alembic_utils.testbase import (
+    TEST_VERSIONS_ROOT,
+    reset_event_listener_registry,
+    run_alembic_command,
+)
+
+# NAME_DEPENDENCIES inc. explicit depends_on
+A = PGView(
+    schema="public",
+    signature="A_view",
+    definition="select 1 as one",
+    depends_on=[],
+)
+
+B_A = PGMaterializedView(
+    schema="public",
+    signature="B_view",
+    definition='select * from public."A_view"',
+    depends_on=[A],
+)
+
+C_A = PGView(
+    schema="public",
+    signature="C_view",
+    definition='select * from public."A_view"',
+    depends_on=[A],
+)
+
+D_B = PGView(
+    schema="public",
+    signature="D_view",
+    definition='select * from public."B_view"',
+    depends_on=[B_A],
+)
+
+E_AD = PGView(
+    schema="public",
+    signature="E_view",
+    definition='select av.one, dv.one as two from public."A_view" as av join public."D_view" as dv on true',
+    depends_on=[A, D_B],
+)
+
+
+def test_create_revision_with_explicit_depends_on(engine) -> None:
+    register_entities([B_A, E_AD, D_B, C_A, A])
+
+    run_alembic_command(
+        engine=engine,
+        command="revision",
+        command_kwargs={"autogenerate": True, "rev_id": "1", "message": "create"},
+    )
+    print(run_alembic_command(engine=engine, command="upgrade", command_kwargs={"revision": "1", "sql": True}))
```

Comment:
I can see in this test file multiple debug prints

Reply:
As agreed left in commented state for now